doc/source/data/batch_inference.rst

```diff
@@ -0,0 +1,752 @@
+    :ref:`advanced guide<batch_inference_advanced_pytorch_example>`.
+
+Batch inference refers to generating model predictions on a set of input data.
+The model can range from a simple a Python function to a complex neural network.
+In batch inference, also known as offline inference, your model is run on a large
+batch of data on demand.
+
+.. margin::
+
+    This is in contrast to online inference, where the model is run immediately on a
+    data point when it becomes available.
+
+Here's a simple schematic of batch inference, "mapping" batches to predictions
+via model inference:
+
+.. figure:: images/batch_inference.png
+
+  Evaluating a batch of input data with a model to get predictions.
+
+Batch inference is a foundational workload for many AI companies, especially since
+more and more pre-trained models become available.
+And while batch inference looks simple at the surface, it can be challenging to do right in production.
+For instance, your data batches can be excessively large, too slow to process sequentially,
+or might need custom preprocessing before being fed into your models.
+To run inference workloads effectively at scale, you need to
+
+- manage your compute infrastructure and cloud clusters
+- parallelize data processing and utilize all your cluster resources (CPUs & GPUs)
+- efficiently transferring data between cloud storage, CPUs for preprocessing, and GPUs for model inference.
+
+Here's a more realistic view of batch inference for modern AI applications:
+
+.. figure:: images/batch_inference_overview.png
+
+  Evaluating a batch of input data with a model to get predictions.
+
+Why use Ray for batch inference?
+---------------------------------
+
+There are many good reasons to use Ray for batch inference today, even if your current
+use case does not require scaling yet:
+
+1. **Faster and Cheaper for modern Deep Learning Applications**: Ray is built for
+    complex workloads and supports loading and preprocessing data with CPUs and model inference on GPUs.
+2. **Cloud, framework, and data format agnostic**. Ray Data works on any cloud provider,
+    any ML framework (like PyTorch and Tensorflow) and does not require a particular file format.
+3. **Out of the box scaling**: The same code that works on one machine also runs on a
+    large cluster without any changes.
+4. **Python first** You can express your inference job directly in Python instead of
+    YAML files or other formats.
+
+.. _batch_inference_quickstart:
+
+Quick Start
+-----------
+
+To get started with batch inference with Ray, the first thing you need to do is
+install Ray together with its data processing library Ray Data:
+
+.. code-block:: bash
+
+    pip install ray[data]
+
+Running batch inference is conceptually easy and requires just three simple steps
+that we'll walk through in this guide:
+
+1. Load your data into a Ray ``Datastream`` and optionally apply any preprocessing you need.
```

Comment:
Consider getting rid of the `Datastream` terminology here and instead do `Load your data and optionally apply any preprocessing you need.`, where `Load you data` links to https://anyscale-ray--34567.com.readthedocs.build/en/34567/data/api/input_output.html#input-output -- this will be far more useful for people to understand :)

Reply:
@pcmoritz well, I agree with you, but then why did we go for this naming convention in the first place? we literally just changed it. I don't see the benefit, yet I'm happy to disagree and commit. But changing it and then _not_ committing fully is the worst of all scenarios.